### PR TITLE
Various travis improvements

### PR DESCRIPTION
**Pull request by dpgeorge**
- only fetch submodules when needed for any given build (saves time, bandwidth)
- build esp8266 as part of Travis CI, takes about 1 minute in total
- build esp32, takes about 3 minutes

### CODE REVIEW

**Comment by nevercast** [comment]
If this is of any help, you're welcome to peruse my Docker container that I use for building ESP32 port MicroPython in CI.

[Dockerfile](https://gist.github.com/nevercast/5843d0171ec6c4bf91965880f89d73f7)
[build.sh](https://gist.github.com/nevercast/ff19b84e31972a6e2f85067b174cd9e0)

File: .travis.yml [hunk 1]
```diff
@@ -125,7 +125,7 @@ jobs:
     - stage: test
       env: NAME="esp32 port build"
       install:
-        - sudo pip install pyparsing
```
Comment (nevercast) [inline .travis.yml]: There is a requirements.txt in ESP-IDF if that matters.

Comment (dpgeorge) [reply .travis.yml]: thanks, I'm trying to optimise the total time, so only installing those things that are necessary

File: .travis.yml [hunk 2]
```diff
@@ -125,7 +125,8 @@ jobs:
     - stage: test
       env: NAME="esp32 port build"
       install:
-        - sudo python3 -m pip install pyparsing
+        - sudo apt-get install python3-pip
+        - sudo pip3 install pyparsing
```
Comment (nevercast) [inline .travis.yml]: You may want to upgrade pip too, I recall having trouble with the distribution provided release of `python3-pip`.
```
apt-get install -y --no-install-recommends python3-pip
python3 -m pip install --upgrade pip setuptools
```

**Comment by dpgeorge** [comment]
Ok, this is good to go in.  Total Travis CI wall-clock time is under 7 minutes including building of esp32 and esp8266.

**Comment by nevercast** [comment]
Nice work, @dpgeorge. The wall clock time for my own build is just under 10 minutes. I might have to make some trimmings, as you have done.

**Comment by dpgeorge** [comment]
Merged, see e06dcad5d385369762f62aeccaff0ae3c6626acf

**Pull request closed without merging**